compiler/codegen/expr.cpp

```diff
@@ -2209,133 +2213,6 @@ GenRet codegenDynamicCastCheck(GenRet cid_Td, Type* C)
-    // in order to put up with structure expansion done by clang
-    // (see canExpandIndirectArgument)
-    // TODO - this should actually depend on the clang ABI,
-    // or else we should find a way to disable the optimization in clang.
-    //   It should be possible to get the necessary information from clang
-    //   with cgModule->getTypes()->arrangeFunctionDeclaration(FunctionDecl)
-
-
-    // Work with a prefix of the structure/vector argument.
-    llvm::Type* int8_type;
-    llvm::Type* int8_ptr_type;
-    llvm::Type* dst_ptr_type;
-
-    llvm::Value* arg_ptr;
-    llvm::Value* arg_i8_ptr;
-    llvm::Value* cur_ptr;
-    llvm::Value* casted_ptr;
-    llvm::Value* cur;
-    int64_t offset = 0;
-    int64_t cur_size = 0;
-    int64_t arg_size = 0;
-    int64_t targ_size = 0;
-
-    int8_type = llvm::Type::getInt8Ty(info->llvmContext);
-    int8_ptr_type = int8_type->getPointerTo();
-
-    arg_size = getTypeSizeInBytes(info->module->getDataLayout(), t);
-    assert(arg_size >= 0);
-    targetType = fnType->getParamType(outArgs.size());
-    targ_size = getTypeSizeInBytes(info->module->getDataLayout(), targetType);
-
-    // Allocate space on the stack...
-    // Some ABIs will increase the size of small structs,
-    // e.g. on aarch64, a struct < 128 bits will be rounded
-    // up to a multiple of 64 bits.
-    if (targ_size > arg_size) {
-      arg_ptr = createVarLLVM(targetType, "");
-      arg_ptr = info->irBuilder->CreatePointerCast(arg_ptr, t->getPointerTo());
-    } else {
-      arg_ptr = createVarLLVM(t, "");
-    }
-    arg_i8_ptr = info->irBuilder->CreatePointerCast(arg_ptr, int8_ptr_type, "");
-
-    // Copy the value to the stack...
-    info->irBuilder->CreateStore(v, arg_ptr);
-
-    while(offset < arg_size) {
-      if( outArgs.size() >= fnType->getNumParams() ) {
-        INT_FATAL("Could not convert arguments for call");
-      }
-      targetType = fnType->getParamType(outArgs.size());
-      dst_ptr_type = targetType->getPointerTo();
-      cur_size = getTypeSizeInBytes(info->module->getDataLayout(), targetType);
-
-      assert(cur_size > 0);
-
-      if (cur_size <= arg_size && offset + cur_size > arg_size) {
-        INT_FATAL("Could not convert arguments for call");
-      }
-
-      // Now load cur_size bytes from pointer into the argument.
-      cur_ptr = info->irBuilder->CreateConstInBoundsGEP1_64(arg_i8_ptr,
-                                                          offset);
-      casted_ptr = info->irBuilder->CreatePointerCast(cur_ptr, dst_ptr_type);
-
-      cur = info->irBuilder->CreateLoad(casted_ptr);
-
-      outArgs.push_back(cur);
-
-      //printf("offset was %i\n", (int) offset);
-      offset = getTypeFieldNext(info->module->getDataLayout(), t,
-                 offset + (arg_size < cur_size ? arg_size : cur_size) - 1);
-      //printf("offset now %i\n", (int) offset);
-    }
-  } else {
-    INT_FATAL("Could not convert arguments for call");
-  }
-}
-
-#endif
-
```

Comment:
Trying to reverse engineer this: we were doing all this stuff with homegrown techniques but now we are relying more on the clang's capabilities?

Reply:
Looking at the relevant issue, that seems to be the case. But in any case, I think the PR message can be more verbose.

Reply:
Yes, that's right. I can add a bit to the PR message.